library/src/main/java/com/bumptech/glide/load/model/ModelEquity.java

```diff
@@ -0,0 +1,16 @@
+package com.bumptech.glide.load.model;
+
+/**
+ * An optional interface that models can implement to handle difference in equity
+ * between the model and the target.
+ */
+public interface ModelEquity {
+
+    /**
+     * Returns {@code true} if this Model is equivalent to the given Model.
```

Comment:
Might be worth an expanded comment on what equivalent to is. 

Maybe 
``` 
Returns {@code true} if this Model produces the same image as the equivalent Model to avoid unnecessarily restarting a Request
```

Reply:
It's more complicated than that :) Even after all the explanations about the need :p 

Equals/HashCode represent the fact that the model produce the same image.
isEquivalentTo is a step further it represent the fact that the model use the same way to produce the same image.

But we need to be sure that users still implements correctly hashCode/Equals for the cache part and the equity of resulting image. Not sure how to properly word all this.

Reply:
One more stab:

----
Returns {@code true} if this model produces the same image using the same mechanism (server, authentication, source etc) as the given model.

<p>Models must also implement {@link #equals()} and {@link #hashCode()} to ensure that caching functions correctly.

<p>If this object returns {@code true} from this method for a given Model, it must also be equal to and have the same hash code as the given model. However, this model may be equal to and have the same hash code as a given model but still return {@code false} from this method.  

<p>This method optionally allows you to differentiate between Models that load the same image via multiple different means. For example one Model might load the image from server A and another model might load the same image from server B. The models must be equal to each other with the same hash code because they load the same image. However two requests made with the different models are not exactly the same because the way the image is loaded will differ.
-----

Just realized in writing this actually that this isn't entirely sufficient. Request deduplication in Engine.java is done based on equals/hashCode. If you start two different requests with Models that return true from equals but false from equivalentTo, the Engine will deduplicate the one that starts second into the first one and only allow the first one to run (the second will simply be notified when the first finishes or fails). 

It doesn't matter if you always cancel the first request before the second, but if you were to do something like start two requests, one for each equal but not equivalent model, simultaneously into two different targets, it wouldn't do quite what you expect.

Reply:
I'm not sure Engine should be touched as the target image is the same. 
So if 2 different models points to the same target image most of the time it's better to not fire 2 network / cache retrieval when only 1 is enough.

Only edge case would be 2 different path for same images, called at the exact same time. But one  path is down and is started just before the other.
This leads to 2 fails instead of 1 fail 1 success.

Maybe this edge case should be addressed, but always using isEquivalentTo would IMO be worse as more request than necessary most of the time.

I don't know yet how engine works, but proper way to address would be to only make first call, then on failure if not Equivalent start second call. (Maybe not possible depending on what data is still available when you currently drop the second query).

Reply:
I forget to tell, but PR updated without touching Engine, tell me if / what you want to do for that part. (Maybe as a second PR).